src/sharedIdGenerator.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Generates the shared Id/Ulid. Id generated is 128 bits with first 48 bits of timestamp followed by 80 bit of randomness.
+ * Spec is available in https://github.com/ulid/spec
+ */
```

Comment:
Does this functionality really need its own file?  The src directory is meant for 'core' files, not something that's dependent on a module.  I realize there are restrictions in our project about module files importing other module files, which is why I was wondering the need for a separate file.  Please let me know your thoughts on this point.

Reply:
In order to have the id generation separated from the other shared id functionality like the sync/decode we have it in a separate file.
Since this sharedIdGenerator.js generates ulid , we could rename it to ulidGenerator.js.

There is no restriction to import anything from src folder but we did notice restriction when we added sharedIdGenerator.js within the userId module and tried to import, we encountered ESLint valid-import restriction

Reply:
@SKOCHERI @jsnellbaker 
I think it should either be renamed to `ulidGenerator.js` or moved into the `sharedIdSystem.js` module...as to which one, I don't have a strong opinion.

If it's in a standalone ulidGenerator.js file, I assume it would only ever be included in a Prebid.js build if a module that uses/imports it (like sharedId) is included in the build, right? In other words, this won't add weight to the prebid build unless it's being used somewhere...?

Reply:
@jsnellbaker Awaiting for your thoughts on renaming sharedIdGenerator.js to ulidGenerator.js

Reply:
I'm trying to gather some additional thoughts from other maintainers of Prebid.js on this overall topic.  I'll follow-up when I can.

Reply:
@jsnellbaker We are inclined to move the sharedIDGenerator.js within the sharedId module itself. I hope that should be ok?

Reply:
@SKOCHERI If you wish to do this change, that's fine.  I think if there's a stronger desire to use this type of encoding down-the-road, we can potentially revisit this idea of having a dedicated file for generating the ID.

Reply:
@jsnellbaker  Have moved the sharedId generation within sharedId submodule. Could you compete your review